plugins/inputs/fluentd/fluentd.go

```diff
@@ -0,0 +1,163 @@
+package fluentd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	measurement  = "fluentd"
+	description  = "Read metrics exposed by fluentd in_monitor plugin"
+	sampleConfig = `
+  ## This plugin reads information exposed by fluentd (using /api/plugins.json endpoint).
+  ##
+  ## Endpoint: 
+  ## - only one URI is allowed
+  ## - https is not supported
+  endpoint = "http://localhost:24220/api/plugins.json"
+  	
+  ## Define which plugins have to be excluded (based on "type" field - e.g. monitor_agent)
+  exclude = [
+	  "monitor_agent",
+	  "dummy",
+  ]
+`
+)
+
+// Fluentd - plugin main structure
+type Fluentd struct {
+	Endpoint string
+	Exclude  []string
+	client   *http.Client
+}
+
+type endpointInfo struct {
+	Payload []pluginData `json:"plugins"`
+}
+
+type pluginData struct {
+	PluginID              string  `json:"plugin_id"`
+	PluginType            string  `json:"type"`
+	PluginCategory        string  `json:"plugin_category"`
+	RetryCount            float64 `json:"retry_count"`
+	BufferQueueLength     float64 `json:"buffer_queue_length"`
+	BufferTotalQueuedSize float64 `json:"buffer_total_queued_size"`
```

Comment:
One difference in this revision of your plugin is that it will report 0 for these fields if they do not exist.  In order to handle this you could have these be `*float64` and test for nil before assigning the field.